quickwit-indexing/src/lib.rs

```diff
@@ -71,3 +80,126 @@ pub async fn index_data(
 pub(crate) fn new_split_id() -> String {
     ulid::Ulid::new().to_string()
 }
+
+/// Fetch source configs from storage for each index id.
+/// The source path must have the form `{index_id}.json`.
+// TODO: accept other format.
+async fn load_sources(source_storage: Arc<dyn Storage>, index_ids: &[String]) -> anyhow::Result<Vec<(&String, SourceConfig)>> {
+    let mut source_configs = vec![];
+    for index_id in index_ids {
+        let source_config_filename = format!("{}.json", index_id);
+        let source_config_path = Path::new(&source_config_filename);
+        if !source_storage.exists(source_config_path).await? {
+            info!(index_id=?index_id, "No source file found for index.");
+            continue;
+        }
+        let source_config_bytes = source_storage.get_all(source_config_path).await?;
+        let source_config: SourceConfig = serde_json::from_slice(&source_config_bytes)
+            .with_context(|| {
+                format!(
+                    "Failed to parse source config file `{}`.",
+                    source_config_path.display()
+                )
+            })?;
+        debug!(index_id=?index_id, "Source config built.");
+        source_configs.push((index_id, source_config));
+    }
+    Ok(source_configs)
+}
+
+/// Start Quickwit indexing server.
+pub async fn serve_indexing_cli(args: ServeIndexingArgs) -> anyhow::Result<()> {
```

Comment:
Do you think it might b possible to move this to quickwit-cli?

I am mostly  uncomfortable to have the warp dependency on quickwit-indexing.